sdk/core/core-http/src/serializer.ts

```diff
@@ -393,7 +397,7 @@ function serializeBase64UrlType(objectName: string, value: any): any {
   return value;
 }
 
-function serializeDateTypes(typeName: string, value: any, objectName: string) {
+function serializeDateTypes(typeName: string, value: any, objectName: string): any {
```

Comment:
can we do better than `any`?  For example, `serializeDateTypes()` likely returns a `string`, or a `number`, or `undefined`?

Similar for other serializeXxxx() methods.

Reply:
This is tricky in some cases, let me go through why.

Let's take `function serializeEnumType(objectName: string, allowedValues: Array<any>, value: any): any {`. This returns value, which is of type `any`. One possible fix would be to make these methods like:

`function serializeEnumType<T>(objectName: string, allowedValues: Array<any>, value: T): T {` so that users of the method would have control of the type they're expecting.

In particular, `serializeDateTypes` can either return a number or a string. I think that same `<T>` approach would be helpful there.

In regards to `serializeCompositeType` and `deserializeCompositeType`, that method alone seems to require hours to narrow it down to strict types without allowing the user to specify their own type.

Please let me know if that seems reasonable.

I was able to make the following methods more strict:

- `function serializeByteArrayType(objectName: string, value: Uint8Array): string {`.
- `function serializeBase64UrlType(objectName: string, value: Uint8Array): string {`.
- `function deserializeDictionaryType(serializer: Serializer, mapper: DictionaryMapper, responseBody: any, objectName: string ): { [key: string]: any }`.

Reply:
I think the problem is that the method reuse the parameter `value: any` and return it.  This particular method could be rewritten to return in each if/else-if branch, then at the end return null or undefined.

Could we log an issue for improvement? I don't feel adding returning type of `any` adds any benefits but we can address it separately.

Reply:
I made this issue to track the fix that you mention here: https://github.com/Azure/azure-sdk-for-js/issues/9674

Please feel free to change that issue at will! Thank you so much for helping me review and improve this PR.